### PR TITLE
 parsing help messages for users uploading datapacks

### PR DESCRIPTION
**Pull request by paologum**
warning messages to help users debug datapack issues

all of these are "non-fatal" to chart generation

may need to parse java console logs for more comprehensive error/warning checks

image

image

image

image

### CODE REVIEW

File: server/src/parse-datapacks.ts [hunk 1]
```diff
@@ -897,24 +1065,44 @@ export function processEvent(line: string, subEventType: SubEventType): SubEvent
...
+  if (tabSeparated.length < 2 || tabSeparated[0]?.trim()) {
+    warnings.push({
+      lineNumber: lineCount,
+      warning: `Line in Point column formatted incorrectly`,
+      message: `This line will be skipped in processing`
+    });
+    return null;
+  }
   const age = parseFloat(tabSeparated[1]!);
   const xVal = parseFloat(tabSeparated[2]!);
```
Comment (jay-jhk-lee) [inline server/src/parse-datapacks.ts:1096]: Is tabSeparated[2] guaranteed here? I don't see anything that checks for it

Comment (paologum) [reply server/src/parse-datapacks.ts:1096]: good catch, will add

**Review by JacquiFreeElectron — COMMENTED** [review]
looks good!

File: server/src/parse-datapacks.ts (hunk 1 above)
Comment (paologum) [reply server/src/parse-datapacks.ts:1096]: ope just kidding, i'm reading my code and i have this blurb below

```
// sometimes they don't exist, contrary to file documentation
  if (!isNaN(xVal)) {
    subPointInfo.xVal = xVal;
  }
```
i also have a test case for this but yeah

**Review by jay-jhk-lee — APPROVED** [review]

**Pull request merged**